servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/CatalogAdlsFileSystemOptions.java

```diff
@@ -15,19 +15,20 @@
  */
 package org.projectnessie.quarkus.config;
 
+import io.smallrye.config.WithConverter;
 import java.time.Duration;
 import java.util.Optional;
 import org.projectnessie.catalog.files.adls.AdlsFileSystemOptions;
+import org.projectnessie.catalog.secrets.BasicCredentials;
+import org.projectnessie.catalog.secrets.TokenSecret;
 
 public interface CatalogAdlsFileSystemOptions extends AdlsFileSystemOptions {
   @Override
-  Optional<String> accountName();
+  Optional<BasicCredentials> account();
```

Comment:
How come a converter is not needed?

Reply:
It's a composite - smallrye-config magic